### PR TITLE
optimize redis Dockerfile

### PR DESCRIPTION
**Pull request by Zoey2936**

### CODE REVIEW

File: Containers/redis/Dockerfile [hunk 1]
```diff
@@ -1,13 +1,12 @@
 # From https://github.com/docker-library/redis/blob/master/7.0/alpine/Dockerfile
 FROM redis:7.0.11-alpine
 
-RUN apk add --no-cache openssl bash
-
-COPY start.sh /usr/bin/
-RUN chmod +x /usr/bin/start.sh
+COPY --chmod=775 start.sh /usr/bin/start.sh
 
+RUN apk add --no-cache openssl bash; \
```
Comment (szaimen) [inline Containers/redis/Dockerfile:6]:
```suggestion
RUN set -ex; \
    apk add --no-cache openssl bash; \
```

File: Containers/redis/Dockerfile [hunk 2]
```diff
@@ -1,13 +1,12 @@
...
+    \
 # Give root a random password
```
Comment (szaimen) [inline Containers/redis/Dockerfile:8]: Is that possible?
```suggestion
    # Give root a random password
```

Comment (Zoey2936) [reply Containers/redis/Dockerfile:8]: yes

File: Containers/redis/Dockerfile [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
 
-COPY start.sh /usr/bin/
-RUN chmod +x /usr/bin/start.sh
-
-# Give root a random password
-RUN echo "root:$(openssl rand -base64 12)" | chpasswd
+RUN set -ex; \
+    apk add --no-cache openssl bash; \
+    \
+    # Give root a random password
```
Comment (szaimen) [inline Containers/redis/Dockerfile:9]: I think it acutally needs to start at the beginning. Sorry for that.
```suggestion
# Give root a random password
```

**Review by szaimen — APPROVED** [review]
LGTM

**Pull request merged**